backend/Controllers/raceResults.js

```diff
@@ -130,9 +146,109 @@ router.get('/race-results', (req, res) => {
       .project({ 'calendar.drivers': 1 })
       .toArray()
       .then((results) => {
-        drivers = results;
+        drivers = results[0].calendar.drivers;
       });
-    return drivers[0];
+    return drivers;
+  };
+
+  const getTeamsList = async () => {
+    let teams;
+    await client
+      .db('RGL')
+      .collection('divisions')
+      .find({
+        division: req.query.division,
+      })
+      .project({ 'calendar.teams': 1 })
+      .toArray()
+      .then((results) => {
+        teams = results[0].calendar.teams;
+      });
+    return teams;
+  };
+
+  const setDriversDetails = async (results) => {
+    let driversList = await getDriversList();
+    let raceFormats = Object.keys(results.calendar.raceFormat);
+
+    for (const raceSession of raceFormats) {
+      let selectedSessionResults =
+        results.calendar.races[0].adjustedResults[raceSession];
+
+      for (const [index, driver] of selectedSessionResults.entries()) {
+        let selectedDriverNick = selectedSessionResults[index].nick;
+
+        driver.fullName = driversList.find(
+          (driver) => driver.nick === selectedDriverNick
+        ).fullName;
+
+        driver.number = driversList.find(
+          (driver) => driver.nick === selectedDriverNick
+        ).number;
+
+        driver.team = driversList.find(
+          (driver) => driver.nick === selectedDriverNick
+        ).team;
+      }
+    }
+
+    return results.calendar.races[0].adjustedResults;
+  };
+
+  const setTeamsDetails = async (results) => {
+    let teamsList = await getTeamsList();
+    let raceFormats = Object.keys(results.calendar.raceFormat);
+
+    for (const raceSession of raceFormats) {
+      let selectedSessionResults =
+        results.calendar.races[0].adjustedResults[raceSession];
+
+      for (const [index, driver] of selectedSessionResults.entries()) {
+        let selectedDriverTeam = selectedSessionResults[index].team;
+
+        driver.teamFullName = teamsList.find(
+          (team) => team.name === selectedDriverTeam
+        ).fullName;
+
+        driver.teamLogo = teamsList.find(
+          (team) => team.name === selectedDriverTeam
+        ).logoUrl;
+
+        driver.teamColour = teamsList.find(
+          (team) => team.name === selectedDriverTeam
+        ).colour;
+      }
+    }
+
+    return results.calendar.races[0].adjustedResults;
+  };
+
+  const setPoints = (results) => {
+    let raceFormats = Object.keys(results.calendar.raceFormat);
+
+    for (const raceSession of raceFormats) {
+      let selectedSessionResults =
+        results.calendar.races[0].adjustedResults[raceSession];
+      let pointsSystem = results.calendar.raceFormat[raceSession].pointsSystem;
+      let topDriversForFastestLap = pointsSystem.fastestLap?.pointsFromTop;
+      let pointsForFastestLap = pointsSystem.fastestLap?.points;
+
+      let fastestOfTheSession = selectedSessionResults
```

Comment:
```suggestion
      let fastestInTheSession = selectedSessionResults
```
To chyba by brzmiało lepiej